docs/userguide/importing.rst

```diff
@@ -0,0 +1,36 @@
+.. _codebases
+
+Importing Parsl apps
+--------------------
+
+It may be convenient to define Parsl apps separately from the definition of the
+:class:`~pars.dataflow.dflow.DataFlowKernel`, or in libraries of apps which are
+intended to be imported by other modules. For this reason, the
+:class:`~parsl.dataflow.dflow.DataFlowKernel` is an optional argument to the
+:func:`~parsl.app.app.App` decorator. If the DFK is not passed to the
+:func:`~parsl.app.app.App` decorator, it must be loaded using
+:meth:`DataFlowKernelLoader.load <parsl.dataflow.dflow.DataFlowKernelLoader.load>` prior
+to importing the app.
+
```

Comment:
From a brief look at the code, it looks like the default-dfk resolution happens at the time of making a call to an `@App` annotated function, not at the point of declaring the `@App`-annotated function; in which case you can declare functions, and initialise later, as long as it's happened before executing. (I haven't actually tried that).

If that's the case, libraries of `@App`s can be imported at the top of a file, like other imports - which I think is a nicer style.

Reply:
Good catch @benclifford-- I made that change (having the resolution executed at calltime instead of declaration time) and forgot to change it in the docs. I'll update this.